lusSTR/annot.py

```diff
@@ -619,13 +636,32 @@ def main(args):
         lus = str_dict[locus]['LUS']
         sec = str_dict[locus]['Sec']
         tert = str_dict[locus]['Tert']
-        str_allele = traditional_str_allele(sequence, no_of_repeat_bases, no_of_sub_bases)
-        if (
-            locus in cannot_split or
-            ((len(sequence) % no_of_repeat_bases != 0) and locus not in must_split)
-           ):
+        foren_5 = str_dict[locus]['Foren_5']
+        foren_3 = str_dict[locus]['Foren_3']
+        power_5 = str_dict[locus]['Power_5']
+        power_3 = str_dict[locus]['Power_3']
```

Comment:
This will fail with an `KeyError` if the corresponding keys/values are not in the dictionary. Most of the main function isn't covered at all in the test suite, so I'm nervous about merging this before it's tested a bit. To the extent that we can refactor large sprawling functions into smaller, limited-scope functions, testing will be easier. I'll see what I can do here.

Reply:
But if the locus isn't in the PowerSeq file, it won't be looking for the key for that particular locus, so would it still fail?

Reply:
That code gets executed for every locus: the main function doesn't check which locus it's processing before running those statements. Unless I'm missing something...

Reply:
I just pushed an update that refactors the code a bit. The tests pass, which I guess means I didn't introduce any syntax errors. But still hard to know if the code's doing the right thing without testing.

Reply:
It iterates through each row and identifies the locus for each sequence and from there pulls the information from the str_dict for that particular locus. Since the PowerSeq files will just not have some loci, it'll skip those and never pull the str_dict info for those loci.

Reply:
Got it. In that case, there's an even more concise way to code that assumption. Pending...